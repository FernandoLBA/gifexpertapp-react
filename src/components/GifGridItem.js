import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( { title, url } ) => {

     return (
          <div className="card animate__animated animate__zoomIn">
              <img src={ url } alt={ title }></img>
              {/* limita el contenido del texto */}
              {/* <p>{ title.substr( 0, 15 ) }...</p> */}
              <p>{ title }</p>
          </div>
     );
};

export default GifGridItem;

GifGridItem.propTypes = {
     title: PropTypes.string.isRequired,
     url: PropTypes.string.isRequired
};
