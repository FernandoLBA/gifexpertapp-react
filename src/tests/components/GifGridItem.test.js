import React from 'react';
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";


describe( 'Pruebas al componente <GifGridItem />', () => {

     const title = 'Un título';
     const url = 'https://localhost/algo.jpg';
     const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );

     test( 'Debe mostrar el componente <GifGridItem /> correctamente', () => {
          
          expect( wrapper ).toMatchSnapshot();
     } );

     test('Debe de tener un párrafo con el title', () => {
          
          const p = wrapper.find( 'p' );
          expect( p.text().trim() ).toBe( title );
     });

     test('El tag img debe tener los atributos url y alt iguales a los props', () => {
          
          const img = wrapper.find( 'img' );
          // prop busca el atributo del tag html, src y alt
          expect( img.prop( 'src' ) ).toBe( url );
          expect( img.prop( 'alt' ) ).toBe( title );
     });

     test('Debe contener animate__zoomIn en el className', () => {
          
          const div = wrapper.find( 'div' );

          // revisa si existe la clase animate__zoomIn en el div
          expect( div.prop( 'className' ).includes( 'animate__zoomIn' ) ).toBe( true );
     });
} );
