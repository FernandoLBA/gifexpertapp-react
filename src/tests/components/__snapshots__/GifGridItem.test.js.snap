// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGridItem /> Debe mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
